### ISSUE TITLE
Azure Sphere SDK/Tools Mac OSX support

### ISSUE THREAD

**Issue by galvesribeiro** [MicrosoftDocs/azure-sphere-issues#191]
Hello folks,

The recent OS/Tools/SDK release (finally) got support to non-Windows systems and VSCode. However, only Linux is supported and the more I read more I see "Mac OSX is not on our roadmap". 

As mentioned on other issues, I've got all the tooling, services, CLI working on OSX by making them to target .Net Core 3.0 even before the Linux support was announced but since it is not an OSS project, we can't make any move or share that code. The only missing piece was the native toolchain which wasn't available. 

So, I've looked on a _viable_ workaround for Mac OSX developers and I finally got it working without breaking any "rules"/license/IP and here is the outcome:

![image](https://user-images.githubusercontent.com/4714040/70390815-525bb580-19ad-11ea-828a-c459f82ff138.png)

To make that workaround, we have two options:

1. Full virtualised Linux environment (i.e. VirtualBox, Parallels, VMWare Fusion) hosting to Azure Sphere SDK;
2. Ubuntu Docker container with the Azure Sphere tools

In both options we would use VSCode Remote extensions to build/run the code from the VM while having the edit/debug experience natively on Mac OSX. Option 2 requires more trickery since USB passthru at Docker for Mac isn't supported on Hyperkit yet.

So, while the native support of Mac OSX is not a priority nor is on the roadmap unfortunately, would you guys accept a PR on the documents explaining how to use the VM and VSCode Remote development extensions and/or Docker to get it to work?

I'm pretty sure there are a lot of people that are suffering the same pain as I did and this workaround, even if it is not ideal (since I hope some day we will get the support natively/officially), would unblock a lot of people and at least make the dev experience way more decent. 

Looking forward to hear from you so I can put the docs PR.

Thanks!

**Comment by AshokPeddakotla-MSFT** [comment]
@galvesribeiro Thanks for bringing this to our attention. We are currently investigating and will update you shortly.

**Comment by AshokPeddakotla-MSFT** [comment]
@pennyo Could you please review this and provide your feedback?

**Comment by galvesribeiro** [comment]
@AshokPeddakotla-MSFT ?

**Comment by richardtaylorrt** [comment]
Hi @galvesribeiro - sorry for the slow reply, and thank you for your offer of help. 

We have been discussing internally.  We'd be happy to take a look at a PR for option #1, running the SDK on a Mac in a VM with USB passthrough.  Please go ahead.

We are also planning to ship our own official container for the SDK, although we don't have a specific date for this yet.  So we're not sure we want to doc option #2 yet until we have this ready and because, as you say, there may be challenges around USB passthrough to the container.

thanks again, Richard

**Comment by galvesribeiro** [comment]
Hey @richardtaylorrt, thanks for getting back. I'll make the PR for the option 1 then.

Reg the container, there are several issues related to Docker and Hyperkit today that block us from use a container and Docker for Mac with USB passthrough. So if I may suggest, it would be better for you to allocate your resources/efforts on bring the SDK natively (without container/VM) on OSX as it is way simpler and doesn't rely on 3rd parties to do it. The problem with the USB on containers goes beyond just Docker... It is related to OSX kernel drivers that are now different after Mac OSX Catelina (10.15+) so I would not recommend try that. There are hacky alternatives using `docker-machine` which still relies on VirtualBox which I've already tried and worked but I would not recommend go that path.

I'll write the docs over the weekend and submit the PR. Thanks!

**Comment by richardtaylorrt** [comment]
Hi again @galvesribeiro - on talking to the team, it sounds like our docs GitHub repo isn't public yet, so you may not be able to submit a PR. If so, please feel free to just suggest some text here, and we'll see if it can be incorporated in a future docs push.  How much text do you expect it to be?  If it's a lot, the team wanted me to set expectations that it may take a while for them to process it!

Of course, you'd also be welcome to share this information elsewhere too: on our [MSDN forum](https://social.msdn.microsoft.com/Forums/en-US/home?forum=azuresphere)or by asking and answering your own question on StackOverlow ([they encourage it](https://stackoverflow.com/help/self-answer)!) with tag '[azuresphere](https://stackoverflow.com/questions/tagged/azuresphere)'

**Comment by AshokPeddakotla-MSFT** [comment]
@galvesribeiro As Richard suggested, please follow the suggestions. 
Since we have not heard back from you we will now proceed to close this thread. If there are further questions regarding this matter, please tag me in your reply. We will gladly reopen the issue and continue the discussion.

**Comment by bobbydharrell** [comment]
@richardtaylorrt or @AshokPeddakotla-MSFT I know this thread is Old. BUt I have an Azure Sphere Starter Kit and was looking for a way to set this up on my Mac. Has any progress been made on getting this running on a Mac? Or do I need to run it from a Windows/Ubuntu VM?

**Comment by lakshmisivareddy** [comment]
Hi @galvesribeiro 
can u please help me with the instructions  on how to setup  dev Environment for azure sphere in Mac OS 
i did find any documentation or instructions  any ware
